packages/dds/tree/src/feature-libraries/sequence-field/utils.ts

```diff
@@ -417,35 +396,14 @@ export function tryExtendMark(
         case "MoveOut":
         case "ReturnFrom": {
             const lhsMoveOut = lhs as MoveOut | ReturnFrom;
+            if (!areMergeableReturnFrom(lhs as ReturnFrom, rhs as ReturnFrom)) {
```

Comment:
We should change the input types of `areMergeableReturnFrom` to be `MoveOut | ReturnFrom` (and find a new name for it) or make a separate `arMergeableMoveIn` function and check the types out here.
Otherwise there's a risk we'll add some extra code to `areMergeableReturnFrom` that will either be wrong or crash for `MoveOut`.